controller/src/main/java/org/jboss/as/controller/access/JmxAction.java

```diff
@@ -28,14 +37,37 @@
  * @author <a href="kabir.khan@jboss.com">Kabir Khan</a>
  */
 public class JmxAction {
+    private static final List<AccessConstraintDefinition> NO_CONSTRAINTS = Collections.emptyList();
+
+    private final boolean isNonFacadeMBeansSensitive;
     private final String methodName;
+    private final String operation;
+    private final String attribute;
+    private final ModelNode currentValue;
     private final Impact impact;
+    private final List<AccessConstraintDefinition> constraints;
 
-    public JmxAction(String methodName, Impact impact) {
+    public JmxAction(String methodName, String operation, Impact impact, boolean isNonFacadeMBeansSensitive) {
+        this(methodName, operation, impact, isNonFacadeMBeansSensitive, null, null);
+    }
+
+    public JmxAction(String methodName, String operation, Impact impact, boolean isNonFacadeMBeansSensitive, Attribute attribute) {
+        this(methodName, operation, impact, isNonFacadeMBeansSensitive, null, attribute);
+    }
+
+    public JmxAction(String methodName, String operation, Impact impact, boolean isNonFacadeMBeansSensitive, List<AccessConstraintDefinition> constraints, Attribute attribute) {
         this.methodName = methodName;
         this.impact = impact;
+        this.operation = operation == null ? "test" : operation;
```

Comment:
Is operation ever different from methodName once ModelControllerMBeanServerPlugin is changed to no longer support authz?

Re: the AccessConstraintDefinitions, same comment as JMXTarget applies.

I think isNonFacadeMBeanSensitive fits better in JMXTarget. It relates to targets more than actions.